src/memory/memoryManager.ts

```diff
@@ -22,7 +24,7 @@ export class MemorySegmentManager {
     address: Relocatable,
     data: MaybeRelocatable[]
   ): Result<Relocatable> {
-    data.forEach((d, index) => {
```

Comment:
Removed the forEach, I know you like it @Eikix but it was actually preventing errors to be propagated.